kcm/package/contents/ui/RulesView.qml

```diff
@@ -20,114 +20,37 @@
-                model: ufwClient.rules()
-                bottomMargin: 48 * 2
-                delegate: RuleListItem {
-                    dropAreasVisible: true
-                    width: listView.width
-                    onMove: function (from, to) {
-                        //                    print("moving ", from, " to ", to)
-                        if (from < to)
-                            to = to - 1
+import org.kde.kirigami 2.4 as Kirigami
 
-                        // Force valid positions
-                        to = Math.max(0, to)
-                        to = Math.min(listView.model.rowCount(), to)
 
-                        if (from !== to) {
-                            //                        listView.model.move(from, to)
-                            ufwClient.moveRule(from, to)
-                        }
-                    }
+import org.nomad.ufw 1.0 as UFW
 
-                    onEdit: function (index) {
-                        var rule = ufwClient.getRule(index)
-                        ruleDetailsLoader.setSource("RuleEdit.qml", {
-                                                        rule: rule,
-                                                        newRule: false,
-                                                        x: 0,
-                                                        y: 0,
-                                                        height: mainWindow.height,
-                                                        width: mainWindow.width
-                                                    })
-                    }
+ColumnLayout {
+    id: mainLayout
+    signal editRuleRequest(var rule)
+    signal newRuleRequest()
 
-                    onRemove: function (index) {
-                        ufwClient.removeRule(index)
-                    }
-                }
+    GlobalRules {
+    }
 
-                section.property: "ipv6"
-                section.criteria: ViewSection.FullString
-                section.delegate: sectionHeading
-            }
+    DynamicFirewallRules {
+        onRequestRuleEdition : {
+            mainLayout.editRule(rule)
         }
+    }
 
-        PlasmaComponents.ToolButton {
-            height: 48
-            iconSource: "list-add"
-            text: i18n("New Rule")
-
-            onClicked: {
-                ruleDetailsLoader.setSource("RuleEdit.qml", {
-                                                newRule: true,
-                                                x: 0,
-                                                y: 0,
-                                                height: mainWindow.height,
-                                                width: mainWindow.width
-                                            })
-            }
-
-            anchors.bottom: parent.bottom
-            anchors.bottomMargin: 12
-            anchors.right: parent.right
-//            anchors.rightMargin: 12
+    QQC2.ToolButton {
+        height: 48
```

Comment:
setting a size in a layout is almost always wrong as it'll get clobbered.

implicitHeight / Layout.preferredHeight works better

We probably don't need it.

Reply:
ok.